src/main/java/me/contaria/seedqueue/gui/wall/SeedQueueWallScreen.java

```diff
@@ -403,6 +402,15 @@ private void updatePreparingPreviews() {
         }
     }
 
+    private int getBackgroundPreviews() {
+        return SeedQueue.config.getBackgroundPreviews().orElseGet(() -> {
+            int mainGroupSize = this.layout.main.size();
+            int preparingGroupSize = Layout.Group.totalSize(this.layout.preparing);
+
+            return Math.min(mainGroupSize + preparingGroupSize, SeedQueue.config.maxCapacity - mainGroupSize);
```

Comment:
I disagree with `mainGroupSize + preparingGroupSize`, that would always create a larger amount of background previews than the size of the preparing group.
When I mentioned considering the preparing group size, I was thinking the formula I wrote down should be the default, but if the preparing group is bigger than that it should use the size of preparing.
Maybe something like this could work `Math.min(Math.max(mainGroupSize, preparingGroupSize), SeedQueue.config.maxCapacity - mainGroupSize)`.
For example, in a scenario with 12 Max Queued, 4 Main, 6 Preparing it would be `Math.min(Math.max(4, 6), 12) => 6`.
Open to other suggestions tho!

Reply:
i agree with you having `mainGroupSize + preparingGroupSize` isn't the right way to go, having more previews than available positions in the preparing group doesn't make sense. i like the alternative you have provided